### ISSUE TITLE
Account settings: "Hide my Gravatar" button is overlapping text

### ISSUE THREAD

**Issue by jjchrisdiehl** [Automattic/wp-calypso#74598]
### Quick summary

The "Hide my photo and Gravatar profile" toggle under https://wordpress.com/me is overlapping its description text, it also makes the toggle difficult to click:

![image](https://user-images.githubusercontent.com/16580129/226036552-c8715629-0ecf-4522-b04f-88462bb87530.png)

Also broken on RTL interfaces.

### Steps to reproduce

1. Go to https://wordpress.com/me
2. Observe the toggle overlapping its text

### What you expected to happen

The toggle should be left align to the description text without overlapping.

### What actually happened

The toggle overlaps its description text.

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple, Atomic

### Logs or notes

Chrome Version 111.0.5563.64 (Official Build) (arm64)

**Comment by jjchrisdiehl** [comment]
Upon further testing, it appears the ToggleControl styling (or maybe its parent BaseControl) breaks when the associated text label gets too long.

Example:

![image](https://user-images.githubusercontent.com/16580129/226045818-043cd2c4-930d-498d-bfb3-f0a17986ad80.png)

![image](https://user-images.githubusercontent.com/16580129/226045676-ff5d2f05-6740-4d4b-b642-2c94f9fa9eeb.png)

**Comment by jjchrisdiehl** [comment]
@mirka 👋  Looks like you worked on a [similar issue](https://github.com/WordPress/gutenberg/commit/2b8a42dd5facce243f8670ba6af7870f16eee722) which aimed to resolve this.

**Comment by mirka** [comment]
Tl;dr — Updating the `@wordpress/components` dependency to 23.0.0 should fix the issue, although I believe this will require React 18.

This issue should be fixed by the upstream fix in https://github.com/WordPress/gutenberg/pull/45962, and theoretically it [should've shipped](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md#2210-2022-11-16) in 22.1.0, which is exactly the [version used](https://github.com/Automattic/wp-calypso/blob/ae067e6882b036254528f65fc8c2635499af42ba/client/package.json#L83) in `wp-calypso/client`. However, for some reason the rendered HTML/CSS on `wordpress.com/me` is of the older ToggleControl before the bug fix. I confirmed in an isolated CodeSandbox that indeed the ToggleControl fix **did not** ship in 22.1.0, and instead is included from 23.0.0.

**Comment by sejas** [comment]
Hi @mirka , @jjchrisdiehl,
I'm onboarding @katinthehatsite to calypso.
Would it be ok if we add a custom CSS to fix this specific toggle until calypso updates the `@wordpress/components` dependency?

**Comment by mirka** [comment]
> Would it be ok if we add a custom CSS to fix this specific toggle until calypso updates the `@wordpress/components` dependency?

Sure, just make sure you remember to remove the quick fix ASAP so it doesn't cause another regression when the dependency is updated 🙂
